x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Aggregate.java

```diff
@@ -132,11 +133,24 @@ public boolean expressionsResolved() {
     @Override
     public List<Attribute> output() {
         if (lazyOutput == null) {
-            lazyOutput = mergeOutputAttributes(Expressions.asAttributes(aggregates()), emptyList());
+            lazyOutput = outputAttributes(aggregates);
         }
         return lazyOutput;
     }
 
+    public static List<Attribute> outputAttributes(List<? extends NamedExpression> aggregates) {
+        return mergeOutputAttributes(Expressions.asAttributes(aggregates), emptyList());
+    }
+
+    @Override
+    protected AttributeSet computeReferences() {
+        return requiredInputAttributes(aggregates, groupings);
+    }
+
+    public static AttributeSet requiredInputAttributes(List<? extends NamedExpression> aggregates, List<? extends Expression> groupings) {
+        return Expressions.references(groupings).combine(Expressions.references(aggregates));
+    }
```

Comment:
Are there any consumer of these methods? I'm not a fan of these cookie cutter method which in practice are one liners.

Reply:
This is re-used in `AggregateExec`; to avoid it from accidentally diverging from `Aggregate`, I factored this out into a helper method.